test/generaloption.py

```diff
@@ -764,6 +773,106 @@ def raise_error(msg, *args):
                               go_args=['--level-level'], envvar_prefix='GENERALOPTIONTEST', error_env_options=True,
                               error_env_option_method=raise_error)
 
+    def _match_testoption1_sysexit(self, args, msg):
+        self.mock_stderr(True)  # reset
+        self.assertErrorRegex(SystemExit, '.*', TestOption1, go_args=args)
+        stderr = self.get_stderr()
+        self.assertTrue(msg in stderr)
```

Comment:
add `self.mock_stderr(False)` here to disable the mocking (so we can actually see something if something is broken hard :)

maybe the `self.assertErrorRegex` during mocking isn't a good idea either, will we see the actual failed test since `stderr` is being mocked?

maybe this is a good place for `try`/`except`/`finally` instead of `self.assertErrorRegex`:

```python
def _match_testoption1_sysexit(self, args, msg):
    """Check whether parsing supplied arguments with TestOption1 results in SystemExit error being raised."""
    system_exit = False
    self.mock_stderr(True)  # reset
    # purposely not using self.assertErrorRegex, since we're mocking stderr...
    try:
        TestOption(go_args=args)
    except SytemExit:
        system_exit = True
    finally:
        self.mock_stderr(False)
    self.assertTrue(system_exit)
```